### ISSUE TITLE
Job is getting stuck

### ISSUE THREAD

**Issue by Raquelie** [rasbt/mlxtend#540]
Hello

I am facing an issue with the sequential feature selector. I am not sure if the issue is with mlxtend or scikit-learn (or something else) but I am giving it a try here.

This is my code, roughly:

```
from sklearn.model_selection import ParameterGrid, GridSearchCV, StratifiedKFold
from mlxtend.feature_selection import SequentialFeatureSelector as SFS
from sklearn import svm
import numpy as np
import pandas as pd

df = pd.read_csv(sourcefile)
Xr_t = df.iloc[:, 0:size] 
yr = df[clas]  
cv = StratifiedKFold(n_splits=5, shuffle=True, random_state=np.random.RandomState(seed=34))
Xr = Xr_t.to_numpy()
param = list(ParameterGrid(parameter_dict))
for p in param:
    est = estimator(**p)
    print(p)
    sfs1 = SFS(estimator=est, k_features=features, forward=fw, pre_dispatch=1, n_jobs=1,
                   floating=True, scoring='roc_auc', verbose=10, cv=cv_pck)
    sfs1 = sfs1.fit(Xr, yr)
```

The estimator is svm.SVC with kernel='linear'. The parameter dict is:
parameters = {'C': [0.0001, 0.001, 0.01, 0.1, 1, 10, 100, 1000, 10000]}

The selection is done well for C <=1000, but when C=10000 it gets stuck. This is, no more output for days, while I can still see a job with a high CPU usage, and I have to manually kill the process since Ctrl-C doesn't do anything (before getting "stuck" it does work).

{'C': 10000, 'kernel': 'linear'}
[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:    0.0s remaining:    0.0s
[Parallel(n_jobs=1)]: Done   2 out of   2 | elapsed:    0.1s remaining:    0.0s
[Parallel(n_jobs=1)]: Done   3 out of   3 | elapsed:    0.3s remaining:    0.0s
[Parallel(n_jobs=1)]: Done   4 out of   4 | elapsed:    0.3s remaining:    0.0s
[Parallel(n_jobs=1)]: Done   5 out of   5 | elapsed:    0.5s remaining:    0.0s
[Parallel(n_jobs=1)]: Done   6 out of   6 | elapsed:    0.6s remaining:    0.0s
[Parallel(n_jobs=1)]: Done   7 out of   7 | elapsed:    1.7s remaining:    0.0s
[Parallel(n_jobs=1)]: Done   8 out of   8 | elapsed:    2.1s remaining:    0.0s
[Parallel(n_jobs=1)]: Done   9 out of   9 | elapsed:    2.3s remaining:    0.0s

Weirdly, if I remove the random state in the StratifiedKFold, it executes and finishes correctly. If I remove that particular parameter it does work too. 

I have the same issue in Win10, Linux and Jupyter notebook with anaconda. I have downgraded to scikit-learn to 19.02 and have the same problem.

Any ideas will be appreciated.

(Apologizes if this is not the right place for a question like this)

**Comment by rasbt** [comment]
Hm, that's weird and definitely shouldn't happen. Based on my teaching experience, there are sometimes unfortunately all kinds of issues with Windows and multiprocessing in Python. It sometimes seems arbitrary and it's hard to pin down.

In this case, it could be something feature-selection related (i.e., that it internally results in a tie or circular behavior during evaluation that causes it to hang) or it could be related to joblib/multiprocessing.

What we could try to fidn out more ...

- can you change the random seed and see if the error still occurs? If it doesn't we know that it might be some issue that is not due to multiprocessing
- is it possible that you can share the exact code and dataset to reproduce the error? I could then run it on macOS and Linux to see if this also occurs on my systems. If yes, we know that it is not-Windows specific

**Comment by Raquelie** [comment]
Hello, 
Thanks for your quick answer. Actually, my bad, I debugged down to the libsvm level and it was a convergence issue (I hadn't set a max number of iterations). However, running the same features (manually), parameters and splits (random seed) but without the feature selection library (this is, the same model that fails to converge) doesn't give any issues, which led me to believe it could be a joblib issue. What's more, if my dictionary of parameters only has one value (the one that fails when there are several), it works well too.
I guess there is a random component somewhere that I am not taking into account. Or something else with my code maybe. 
Anyway, thanks again, we can close this :)

**Comment by rasbt** [comment]
glad to hear that you found the culprit and that it likely wasn't due to a bug in mlxtend. In case you notice some other weird behavior, please feel free to reopen this!
